src/System.Private.ServiceModel/tests/Stress/SharedPoolsOfWCFObjects/CallStats.cs

```diff
@@ -0,0 +1,205 @@
+                SunnyDay.AddCallTiming(sw.ElapsedTicks);
+            }
+            catch (Exception e)
+            {
+                RainyDay.AddCallTiming(sw.ElapsedTicks);
+            }
+        }
+
+        public T CallFuncAndRecordStats<T>(Func<T> func)
+        {
+            T result = default(T);
+            Stopwatch sw = new Stopwatch();
+            try
+            {
+                sw.Restart();
+                result = func();
+                SunnyDay.AddCallTiming(sw.ElapsedTicks);
+            }
+            catch (Exception e)
+            {
+                RainyDay.AddCallTiming(sw.ElapsedTicks);
+            }
+            return result;
+        }
+
+        public T CallFuncAndRecordStats<T, P>(Func<P, T> func, P param)
+        {
+            T result = default(T);
+            Stopwatch sw = new Stopwatch();
+            try
+            {
+                sw.Restart();
+                result = func(param);
+                SunnyDay.AddCallTiming(sw.ElapsedTicks);
+            }
+            catch (Exception e)
+            {
+                RainyDay.AddCallTiming(sw.ElapsedTicks);
+            }
+            return result;
+        }
+
+        public async Task CallAsyncFuncAndRecordStatsAsync(Func<Task> func)
+        {
+            Stopwatch sw = new Stopwatch();
+            try
+            {
+                sw.Restart();
+                await func();
+                SunnyDay.AddCallTiming(sw.ElapsedTicks);
+            }
+            catch (Exception e)
+            {
+                RainyDay.AddCallTiming(sw.ElapsedTicks);
+            }
+        }
+
+        public async Task CallAsyncFuncAndRecordStatsAsync<P>(Func<P, Task> func, P param)
+        {
+            Stopwatch sw = new Stopwatch();
+            try
+            {
+                sw.Restart();
+                await func(param);
+                SunnyDay.AddCallTiming(sw.ElapsedTicks);
+            }
+            catch (Exception e)
+            {
+                RainyDay.AddCallTiming(sw.ElapsedTicks);
+            }
+        }
+    }
+
+    public class TimingPercentileStats
+    {
+        public DateTime Time;       // Time when the stats were calculated
+        public int[] Centile;       // negative % means take bottom part, positive % means take top part
+        public long[] AvgTiming;    // within this time
+    }
+
+    public class CallTimingStatsCollector
+    {
+        private int _samples;
+        private int _numCalls;
+        private long[] _timings;
+        private Object _lock = new Object();
+        private Func<long[], TimingPercentileStats> _statsCalculator;
+        public ConcurrentQueue<TimingPercentileStats> PercentileStats { get; set; }
+
+        public CallTimingStatsCollector(int samples, Func<long[], TimingPercentileStats> statsCalculator)
+        {
+            _samples = samples;
+            _timings = new long[_samples * 2];
+            _statsCalculator = statsCalculator;
```

Comment:
No changed needed in this PR, but I'm a little confused by the "_samples * 2" length for timings and the index modulo computations below.  Consider for a future PR comments saying why _timings is twice as long as _samples and why the modulo computations.  I presume the _timings[] array is just a circular buffer that captures the most recent timings, not all timings.